qutip/lattice.py

```diff
@@ -0,0 +1,785 @@
+            raise Exception("\n\n inter_hop need to be a Qobj.")
+        else:
+            if ( inter_hop.dims[1][0] == 0):
+                if (cell_num_site == 1):
+                    siteT = Qobj([[-1]])
+                else:
+                    siteT = basis(cell_num_site,cell_num_site-1)*basis(cell_num_site,0).dag() + basis(cell_num_site,0)*basis(cell_num_site,cell_num_site-1).dag()
+                inter_hop = tensor( Qobj(siteT), qeye(self.cell_site_dof) )                                
+                nSi = inter_hop.shape
+                if (nSb != nSi):
+                    raise Exception("inter_hop is required to have the same \
+                    dimensionality as cell_Hamiltonian.")                
+                self._H_inter_list = [inter_hop]     
+
+        self._is_consistent = self._checks(check_if_complete = True)
+
+    def __repr__(self):
+        s = ""
+        s += ("Lattice1d object: " +
+              "Number of cells = " + str(self.num_cell) +
+              ",\nNumber of sites in the cell = " + str(self.cell_num_site) +
+              ",\nDegrees of freedom per site = " + str(self.lattice_tensor_config[2:len(self.lattice_tensor_config)]) +              
+              ",\nLattice tensor configuration = " + str(self.lattice_tensor_config) +
+              ",\nbasis_Hamiltonian = " + str(self._H_intra) +              
+              ",\ninter_hop = " + str(self._H_inter_list) +
+              ",\ninter_hop = " + str(self.cell_tensor_config) +
+              ", isherm = " + str(self.cell_tensor_config) +
+              "\n")
+        if (self.PBCx == 1):
+            s += "Boundary Condition:  Periodic"
+        else:
+            s += "Boundary Condition:  Hardwall"            
+
+        return s
+    
+    def _checks(self,check_if_complete = False):
+        """
+        All user inputs are checked in __init__() at the time of initialization.
+        Here it is checked if all the different inputs are consistent with each
+        other. If it is called with check_if_complete == True, it returns if 
+        the Lattice1d instance is complete or not.
+        Returns
+        -------
+        is_consistent : bool
+            Returns True if all the attributes and parameters are consistent
+            for a complete definition of Lattice1d and False otherwise.        
+        """                 
+        if (not isherm(self._H_intra) ):
+            raise Exception(" cell_Hamiltonian is required to be Hermitian. ");
+
+        nRi = self._H_intra.shape[0]        
+        if ( nRi != self._length_of_unit_cell):
+            stt = " cell_Hamiltonian is not dimensionally consistent with \
+            cell_num_site and cell_site_dof. cell_structure() function can be \
+            used to obtain a valid structure for the cell_Hamiltonian and \
+            inter_hop."
+            raise Exception("stt");            
+
+        is_consistent = True
+        return is_consistent        
```

Comment:
Check should not return always `True`. If you want it to raise and error when something is wrong, simply return nothing.